pkg/actions/nodegroup/upgrade_test.go

```diff
@@ -121,55 +122,181 @@ var _ = Describe("Upgrade", func() {
+					Expect(fakeStackManager.GetManagedNodeGroupTemplateCallCount()).To(Equal(1))
+					Expect(fakeStackManager.GetManagedNodeGroupTemplateArgsForCall(0)).To(Equal(ngName))
+					Expect(fakeStackManager.UpdateNodeGroupStackCallCount()).To(Equal(1))
+					By("upgrading the ReleaseVersion and not updating the ForceUpdateEnabled setting")
+					ng, template, wait := fakeStackManager.UpdateNodeGroupStackArgsForCall(0)
+					Expect(ng).To(Equal(ngName))
+					Expect(template).To(Equal(al2FullyUpdatedTemplate))
+					Expect(wait).To(BeTrue())
+				})
+			})
+		})
+
+		When("ForceUpdateEnabled is set to true but the desired value is false", func() {
+			When("it uses bottlerocket", func() {
+				BeforeEach(func() {
+					fakeStackManager.ListNodeGroupStacksReturns([]manager.NodeGroupStack{{NodeGroupName: ngName}}, nil)
+
+					fakeStackManager.GetManagedNodeGroupTemplateReturns(brForceTrueTemplate, nil)
+
+					fakeStackManager.DescribeNodeGroupStackReturns(&manager.Stack{
+						Tags: []*cloudformation.Tag{
+							{
+								Key:   aws.String(api.EksctlVersionTag),
+								Value: aws.String(version.GetVersion()),
+							},
+						},
+					}, nil)
+
+					fakeStackManager.UpdateNodeGroupStackReturns(nil)
+
+					p.MockEKS().On("DescribeNodegroup", &awseks.DescribeNodegroupInput{
+						ClusterName:   aws.String(clusterName),
+						NodegroupName: aws.String(ngName),
+					}).Return(&awseks.DescribeNodegroupOutput{
+						Nodegroup: &awseks.Nodegroup{
+							NodegroupName:  aws.String(ngName),
+							ClusterName:    aws.String(clusterName),
+							Status:         aws.String("my-status"),
+							AmiType:        aws.String("BOTTLEROCKET_x86_64"),
+							Version:        aws.String("1.20"),
+							ReleaseVersion: aws.String("1.20-20201212"),
+						},
+					}, nil)
+
+					p.MockSSM().On("GetParameter", &ssm.GetParameterInput{
+						Name: aws.String("/aws/service/bottlerocket/aws-k8s-1.21/x86_64/latest/image_version"),
+					}).Return(&ssm.GetParameterOutput{
+						Parameter: &ssm.Parameter{
+							Value: aws.String("1.5.2-1602f3a8"),
+						},
+					}, nil)
+				})
+
+				It("upgrades the nodegroup updating the stack with the kubernetes version", func() {
+					Expect(m.Upgrade(options)).To(Succeed())
+					Expect(fakeStackManager.GetManagedNodeGroupTemplateCallCount()).To(Equal(1))
+					Expect(fakeStackManager.GetManagedNodeGroupTemplateArgsForCall(0)).To(Equal(ngName))
+
+					By("upgrading the ForceUpdateEnabled setting first")
+					Expect(fakeStackManager.UpdateNodeGroupStackCallCount()).To(Equal(2))
+					ng, template, wait := fakeStackManager.UpdateNodeGroupStackArgsForCall(0)
+					Expect(ng).To(Equal(ngName))
+					Expect(template).To(Equal(brForceFalseTemplate))
+					Expect(wait).To(BeTrue())
+
+					By("upgrading the Version next")
+					ng, template, wait = fakeStackManager.UpdateNodeGroupStackArgsForCall(1)
+					Expect(ng).To(Equal(ngName))
+					Expect(template).To(Equal(brFulllyUpdatedTemplate))
+					Expect(wait).To(BeTrue())
```

Comment:
I have to read through this again, but there seem to be a lot of duplicate checks around here. I do have table tests in ginkgo though because they are freakishly unintuitive to read, so I don't know yet how I like this. :D

Reply:
I'm not really sure how  I could of improve it